specification/under_development/Instruction_Set.md

```diff
@@ -0,0 +1,52 @@
+# Instruction Set
+
+A backend that supports quantum computations is composed of multiple processing units that include one or more quantum chips as well as classic computing resources.
+A quantum program is a combination of instructions that execute on a quantum processor, 
+instructions that execute on an adjacent classical processor, and data transfer between processors.
+
+The QIR profile defines which instructions are used/supported for classical computations and data transfer, while the Quantum Instruction Set (QIS) defines which instructions are supported by the quantum processor. The profile and QIS can be defined/chosen mostly independently, with the caveat that a profile specification may contain restrictions that need to be satisfied for an QIS to be compatible (compliant) with that profile.
+To target a program to a specific backend hence requires selecting both a profiles and QIS that is supported by that backend, see also [Compilation and Targeting](Compilation_And_Targeting.md). 
+
+## Runtime Functions
+
+Instructions that execute on a classical processor or serve data transfer are expressed as LLVM instructions, including calls to `__quantum__rt__*` functions.
```

Comment:
I think it would be good to make this more explicit: `__quantum__rt__*`, `__quantum__trans_to_cpu__*`, `__quantum__trans_to_qpu__*` and `__quantum__qis__*`. This would leave no ambiguity about where the function should live.

Reply:
This might become messy when we drop the simplifying pretense that there are only two processors - a cpu and a qpu. 1) the classical processor is not necessarily a cpu, 2) we are ultimately looking at highly complex multi-processor systems that all look very differently. Hence, I am hesitant to encode too much detail in the naming.